Sloth.Core/Extensions/StringExtensions.cs

```diff
@@ -19,5 +21,39 @@ public static string Base64Decode(this string base64EncodedData)
             var base64EncodedBytes = System.Convert.FromBase64String(base64EncodedData);
             return System.Text.Encoding.UTF8.GetString(base64EncodedBytes);
         }
+
+        public static string SafeTruncate(this string value, int max)
+        {
+            if (string.IsNullOrWhiteSpace(value) || value.Length <= max)
+            {
+                return value;
+            }
+
+            if (max <= 0)
+            {
+                throw new ArgumentOutOfRangeException();
+            }
+
+            return value.Substring(0, max);
```

Comment:
@jSylvestre I think you can just replace this whole thing by `return value?.Substring(0,max);`. pretty sure substring doesn't care if you ask for max > length, and the `?` will handle the null case.